### PR TITLE
Timeout for socket connection

### PR DESCRIPTION
**Pull request by esnoeijs**
Added timeout to the socket connection as well as some exceptions for failures.

I needed this to ensure that network issues or faulty configuration don't just hang the process when trying to send.

### CODE REVIEW

File: src/Zabbix/Sender.php [hunk 1]
```diff
@@ -32,16 +32,25 @@ class Sender
...
      * Zabbix constructor.
      *
-     * @param string $server Zabbix host
-     * @param int $port Zabbix port
+     * @param string $server  Zabbix host
+     * @param int    $port    Zabbix port
+     * @param float  $timeout Connection timeout 1 second by default
      */
-    public function __construct($server, $port = 10051)
+    public function __construct($server, $port = 10051, $timeout = 1.0)
```
Comment (disc) [inline src/Zabbix/Sender.php]: Lets use zero timeout for backward compatibility.
You can override default value with 1.0 sec in your application

File: src/Zabbix/Sender.php [hunk 2]
```diff
@@ -132,10 +141,24 @@ protected function buildRequestBody()
     protected function sendData($body)
     {
         $socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
+        if (!$socket) {
+            throw new \RuntimeException(socket_strerror(socket_last_error()));
+        }
+
+        // Set send and receive timeout
+        $timeoutSettings = ['sec' => floor($this->timeout), 'usec' => ($this->timeout - floor($this->timeout)) * 1e6];
```
Comment (disc) [inline src/Zabbix/Sender.php:149]: Apply timeout setting if timeout is not empty

**Review by disc — CHANGES_REQUESTED** [review]

**Comment by disc** [comment]
@esnoeijs, thanks for your contribution. There are some small comments about backward compatibility. Anyway, nice job!

File: src/Zabbix/Sender.php (hunk 2 above)
Comment (esnoeijs) [reply src/Zabbix/Sender.php:149]: When SO_SNDTIMEO or SO_RCVTIMEO gets set to 0 it's the equivalent of infinite, which is the default behaviour. 
(https://linux.die.net/man/7/socket)
I've added some type casting in the constructor just to be explicit.

**Review by disc — APPROVED** [review]

**Pull request merged**